DogeNet.BLL/Features/Messages/SendMessage/SendMessageHandler.cs

```diff
@@ -0,0 +1,41 @@
+﻿// <copyright file="SendMessageHandler.cs" company="Leobit">
+// Copyright (c) Leobit. All rights reserved.
+// </copyright>
+
+namespace DogeNet.BLL.Features.Messages.SendMessage
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Text;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using AutoMapper;
+    using DogeNet.DAL;
+    using DogeNet.DAL.Models;
+    using MediatR;
+
+    public class SendMessageHandler : IRequestHandler<SendMessageCommand>
+    {
+        private readonly DBContext context;
+
+        private readonly IMapper mapper;
+
+        public SendMessageHandler(DBContext context, IMapper mapper)
+        {
+            this.context = context;
+            this.mapper = mapper;
+        }
+
+        public async Task<Unit> Handle(SendMessageCommand request, CancellationToken cancellationToken)
+        {
+            var message = this.mapper.Map<Message>(request.model);
+            message.CreatedAt = DateTime.Now;
+
+            await this.context.Messages.AddAsync(message);
+            this.context.SaveChanges();
```

Comment:
await saving instead of adding?

Reply:
![image](https://user-images.githubusercontent.com/77879337/145009587-18671720-a73c-46e1-a746-2535716e9a63.png)
done